.gitlab/llnl-ci-conda.yml

```diff
@@ -0,0 +1,115 @@
+variables:
+  CONDA_WORK_DIR: $SDK_WORK_DIR/conda
+  RADICAL_PILOT_DBURL: ${MONGODB_CONNECTION_STRING}?tlsAllowInvalidCertificates=true
+  TEST_TYPE: "conda"
+
+stages:
+  - setup
+  - test
+  - cleanup
+
+.final_steps:
+  script: &finalize
+    - chgrp -fR exaworks $CONDA_WORK_DIR || true
+    - chmod -fR 02770 $CONDA_WORK_DIR    || true
+
+conda-env-setup:
+  stage: setup
+  parallel:
+    matrix:
+      - HOST : [ "quartz" ]
+        CONDA_ENV_NAME: [ "quartz-env" ]
+        RUNNER_TYPE: "shell"
+      - HOST : [ "ruby" ]
+        CONDA_ENV_NAME: [ "ruby-env" ]
+        RUNNER_TYPE: "shell"
+      # - HOST : [ "lassen" ]
+      #   ENVIRONMENT: [ "BATCH" ]
+      #   CONDA_ENV_NAME: [ "lassen-env" ]
+      #   RUNNER_TYPE: "batch"
+  tags:
+    - $HOST
+    - $RUNNER_TYPE
+  script:
+    - export REQUIREMENTS_DIR=$(pwd)
+    - mkdir -p $CONDA_WORK_DIR
+    - cd $CONDA_WORK_DIR
+    - eval "$(conda shell.posix hook)"
+    - conda create --prefix ./${CONDA_ENV_NAME} python=3.7 -y
+    - conda env list
+    - conda activate ./${CONDA_ENV_NAME}
+    - conda env list
+    - ls
+    - pwd 
+    - conda env update --file $REQUIREMENTS_DIR/conda-requirements.yml
+    - conda list
+    - pip list
+    - *finalize
+
+conda-tests:
+  stage: test
+  needs: [ "conda-env-setup" ]
+  parallel:
+    matrix:
+      - HOST : [ "quartz" ]
+        CONDA_ENV_NAME: [ "quartz-env" ]
+        RUNNER_TYPE: "shell"
+      - HOST : [ "ruby" ]
+        CONDA_ENV_NAME: [ "ruby-env" ]
+        RUNNER_TYPE: "shell"
+      # - HOST : [ "lassen" ]
+      #   ENVIRONMENT: [ "BATCH" ]
+      #   CONDA_ENV_NAME: [ "lassen-env" ]
+      #   RUNNER_TYPE: "batch"
+  tags:
+    - $HOST
+    - $RUNNER_TYPE
+  script:
+    - cd $CONDA_WORK_DIR
+    - pwd 
+    - ls
+    - eval "$(conda shell.posix hook)"
+    - echo $(pwd)/${HOST}-exaworks-sdk-venv
+    - echo $CONDA_DEFAULT_ENV
+    - conda activate ./${CONDA_ENV_NAME}
+    - echo $CONDA_DEFAULT_ENV
+    - conda info --envs
+    - conda list
+    - cd SDK
```

Comment:
no need to change directories
```
    - eval "$(conda shell.posix hook)"
    - conda activate $CONDA_WORK_DIR/$CONDA_ENV_NAME
```

same approach for pip-config

Reply:
Done, the environment setup and tests should now all be running relative to the `SDK` pipeline directory. 

This is the virtual environment implementation, but I also used the same premise for the conda env:
```
python3 -m venv ${PIP_WORK_DIR}/${VENV_ENV_NAME}                                                                  
source ${PIP_WORK_DIR}/${VENV_ENV_NAME}/bin/activate
```

Moreover, I added `<env type>_REQUIREMENTS` to the `variable` section to specify the environment's requirements file, so the installation of the requirements depends on just that one variable.